chapters/lexicalstructure.tex

```diff
@@ -250,7 +250,7 @@ \subsection{Strings}\label{strings}
 \subsection{Units of Literal Constants}\label{units-literal-constants}
 
 The following rules apply:
-- The result of `x + L`, `L + x`, `x - L`, `L - x`, `x*L`, `L*x`, `x/L`, where `x` is a variable with non-empty \lstinline!unit! attribute string `<s>` and `L` is an expression containing only literal constants, shall have the same \lstinline!unit! attribute string of `x`.
+- The result of `x + L`, `L + x`, `x - L`, `L - x`, `x*L`, `L*x`, `x/L`, where `x` is an expression with non-empty \lstinline!unit! attribute string `<s>` and `L` is an expression containing only literal constants, shall have the same \lstinline!unit! attribute string of `x`.
```

Comment:
Maybe I'm off-base or too precise again, but I don't think *expressions* even have a `unit` attribute string -- only `Real` does (4.8.1).

Reply:
We could write "real expression". Note that the literal can also be an integer literal, but the assumption is that both x and x+L (etc) are "real expressions".
Alternatively, we could view it as expressions having a unit - in general - but that may require more work.

Reply:
> We could write "real expression".
What about arrays?

Reply:
> > We could write "real expression".
> 
> What about arrays?

That's messier - let's focus on their scalar elements first. 

Usually, arrays are homogenous - i.e., all elements have the same unit - but sometimes they are not (especially when applying filtering to an array).

Note that you can write `Real x[2](unit={"m","m/s"})`, so you can create non-homogenous arrays - but it's not common.
(I believe we added support for that in Dymola in 2013.)